internals/cli/cmd_push.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/canonical/go-flags"
+
+	"github.com/canonical/pebble/client"
+)
+
+const cmdPushSummary = "Transfer a file to the remote system"
+const cmdPushDescription = `
+The push command transfers a file to the remote system.
+`
+
+type cmdPush struct {
+	client *client.Client
+
+	Parents bool   `short:"p"`
+	Mode    string `short:"m"`
```

Comment:
@anpep FYI, I changed these fields names to `Parents` and `Mode` so they matched the 'p' and 'm' (before it was confusing because **M**akeDirs went with 'p' and **P**ermissions went with 'm'). I've done the similar change in `cmd_mkdir.go`.

I realize this makes the names different from the client field names, but that's okay.